libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -974,6 +974,9 @@ void CompilerUtils::convertType(
 				}
 				else
 				{
+					if (auto baseType = dynamic_cast<ArrayType const*>(typeOnStack.baseType()))
+						solUnimplementedAssert(typeOnStack.isDynamicallySized() || !typeOnStack.isDynamicallyEncoded() || !baseType->isByteArray(), "Copying nested calldata dynamic arrays from calldata to memory is not implemented in the old codegen.");
+
```

Comment:
Is it possible to add location here?

Reply:
to the assert? No, those are more lower-level errors, not proper error reportings.